### ISSUE TITLE
Ultralytics Python library ignores v12 weights and downloads v11n every time.

### ISSUE THREAD

**Issue by hack-r** [ultralytics/ultralytics#19938]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

I run the following YOLO code with extra large v12 weights, yet it seems to download the yolov11n model for some reason and uses that. I'm trying to understand this behavior, as I don't see it in the [documentation][1].

    from ultralytics import YOLO
    import torch
    
    torch.cuda.empty_cache()
    
    model = YOLO("/content/yolo12x.pt") # also tried yolo12l.pt
    
    mydata = '/content/yolo_model_definition.yaml'
    
    results = model.train(
        amp=True,
        batch=256,
        cache='disk',
        cos_lr=True,
        data=mydata,
        deterministic=False,
        dropout=0.1,
        epochs=100,
        exist_ok=True,
        fraction=1.0,
        freeze=[0, 1, 2, 3, 4],
        imgsz=224,
        lr0=0.001,
        lrf=0.0001,
        mask_ratio=4,
        multi_scale=True,
        nbs=512,
        optimizer='auto',
        patience=15,
        plots=True,
        pretrained=True,
        project='aihardware',
        name='fine_tune_run',
        rect=False,
        resume=False,
        seed=42,
        val=True,
        verbose=True,
        weight_decay=0.0005,
        warmup_bias_lr=0.1,
        warmup_epochs=3.0,
        warmup_momentum=0.8,
        close_mosaic=10,
        cls=1.0,
        pose=0,
        overlap_mask=True
    )
    
    print(results)

Here's some of the output:

    Freezing layer 'model.4.m.1.m.1.cv1.bn.weight'
    Freezing layer 'model.4.m.1.m.1.cv1.bn.bias'
    Freezing layer 'model.4.m.1.m.1.cv2.conv.weight'
    Freezing layer 'model.4.m.1.m.1.cv2.bn.weight'
    Freezing layer 'model.4.m.1.m.1.cv2.bn.bias'
    Freezing layer 'model.21.dfl.conv.weight'
    AMP: running Automatic Mixed Precision (AMP) checks...
    Downloading https://github.com/ultralytics/assets/releases/download/v8.3.0/yolo11n.pt to 'yolo11n.pt'...
    100%|██████████| 5.35M/5.35M [00:00<00:00, 396MB/s]
    AMP: checks passed ✅
    train: Scanning /content/new/labels.cache... 739 images, 84 backgrounds, 0 corrupt: 100%|██████████| 739/739 [00:00<?, ?it/s]
    train: Caching images (0.1GB Disk): 100%|██████████| 739/739 [00:00<00:00, 59497.67it/s]
    albumentations: Blur(p=0.01, blur_limit=(3, 7)), MedianBlur(p=0.01, blur_limit=(3, 7)), ToGray(p=0.01, num_output_channels=3, method='weighted_average'), CLAHE(p=0.01, clip_limit=(1.0, 4.0), tile_grid_size=(8, 8))
    val: Scanning /content/val/labels... 141 images, 16 backgrounds, 0 corrupt: 100%|██████████| 141/141 [00:00<00:00, 998.33it/s]val: New cache created: /content/val/labels.cache
    
    val: Caching images (0.0GB Disk): 100%|██████████| 141/141 [00:00<00:00, 5131.07it/s]
    Plotting labels to aihardware/fine_tune_run/labels.jpg... 
    optimizer: 'optimizer=auto' found, ignoring 'lr0=0.001' and 'momentum=0.937' and determining best 'optimizer', 'lr0' and 'momentum' automatically... 
    optimizer: AdamW(lr=0.000833, momentum=0.9) with parameter groups 205 weight(decay=0.0), 214 weight(decay=0.0005), 211 bias(decay=0.0)
    TensorBoard: model graph visualization added ✅
    Image sizes 224 train, 224 val
    Using 8 dataloader workers
    Logging results to aihardware/fine_tune_run
    Starting training for 100 epochs...

Of course, the part of that I'm concerned about is:

>     Downloading https://github.com/ultralytics/assets/releases/download/
>     v8.3.0/yolo11n.pt to 'yolo11n.pt'...
>     100%|██████████| 5.35M/5.35M [00:00<00:00, 396MB/s]

I have also tried changing `pretrained` from `True` to a string path to the weights (not that it should need it, since the weights were specified in the creation of the model object). The problem does not happen if I use the YOLO CLI instead of Python:

    !yolo train model=/content/yolo12l.pt \
        data=/content/yolo_model_definition.yaml \
        epochs=100 \
        batch=32 \
        imgsz=224 \
        lr0=0.001 \
        lrf=0.0001 \
        weight_decay=0.0005 \
        dropout=0.1 \
        multi_scale=True \
        optimizer=auto \
        project=aihardware \
        name=fine_tune_run \
        exist_ok=True \
        cos_lr=True \
        cache=disk \
        val=True \
        plots=True \
        seed=42 \
        warmup_epochs=3.0 \
        warmup_bias_lr=0.1 \
        warmup_momentum=0.8 \
        patience=15 \
        cls=1.0 \
        mask_ratio=4 \
        close_mosaic=10 \
        overlap_mask=True \
        freeze=0,1,2,3,4 \
        device=0 \
        amp=True \
        fraction=1.0 \
        verbose=True

The code above uses the v12 weights and does not download v11. However, `ultralytics` seems to be the current version, 8.3.99, so I don't understand the behavior in Python.

  [1]: https://docs.ultralytics.com/modes/train/#train-settings

This issue has also been [documented on Stack Overflow](https://stackoverflow.com/questions/79545081/why-does-yolo-download-a-nano-v11-model-yolov11n-when-given-the-extra-larger-v).

### Environment

Ultralytics 8.3.99 🚀 Python-3.11.11 torch-2.6.0+cu124 CUDA:0 (NVIDIA A100-SXM4-40GB, 40507MiB)
Setup complete ✅ (12 CPUs, 83.5 GB RAM, 42.9/235.7 GB disk)

OS                  Linux-6.1.85+-x86_64-with-glibc2.35
Environment         Colab
Python              3.11.11
Install             pip
Path                /usr/local/lib/python3.11/dist-packages/ultralytics
RAM                 83.48 GB
Disk                42.9/235.7 GB
CPU                 Intel Xeon 2.20GHz
CPU count           12
GPU                 NVIDIA A100-SXM4-40GB, 40507MiB
GPU count           1
CUDA                12.4

numpy               ✅ 2.0.2<=2.1.1,>=1.23.0
matplotlib          ✅ 3.10.0>=3.3.0
opencv-python       ✅ 4.11.0.86>=4.6.0
pillow              ✅ 11.1.0>=7.1.2
pyyaml              ✅ 6.0.2>=5.3.1
requests            ✅ 2.32.3>=2.23.0
scipy               ✅ 1.14.1>=1.4.1
torch               ✅ 2.6.0+cu124>=1.8.0
torch               ✅ 2.6.0+cu124!=2.4.0,>=1.8.0; sys_platform == "win32"
torchvision         ✅ 0.21.0+cu124>=0.9.0
tqdm                ✅ 4.67.1>=4.64.0
psutil              ✅ 5.9.5
py-cpuinfo          ✅ 9.0.0
pandas              ✅ 2.2.2>=1.1.4
seaborn             ✅ 0.13.2>=0.11.0
ultralytics-thop    ✅ 2.0.14>=2.0.0
{'OS': 'Linux-6.1.85+-x86_64-with-glibc2.35',
 'Environment': 'Colab',
 'Python': '3.11.11',
 'Install': 'pip',
 'Path': '/usr/local/lib/python3.11/dist-packages/ultralytics',
 'RAM': '83.48 GB',
 'Disk': '42.9/235.7 GB',
 'CPU': 'Intel Xeon 2.20GHz',
 'CPU count': 12,
 'GPU': 'NVIDIA A100-SXM4-40GB, 40507MiB',
 'GPU count': 1,
 'CUDA': '12.4',
 'Package Info': {'numpy': '✅ 2.0.2<=2.1.1,>=1.23.0',
  'matplotlib': '✅ 3.10.0>=3.3.0',
  'opencv-python': '✅ 4.11.0.86>=4.6.0',
  'pillow': '✅ 11.1.0>=7.1.2',
  'pyyaml': '✅ 6.0.2>=5.3.1',
  'requests': '✅ 2.32.3>=2.23.0',
  'scipy': '✅ 1.14.1>=1.4.1',
  'torch': '✅ 2.6.0+cu124!=2.4.0,>=1.8.0; sys_platform == "win32"',
  'torchvision': '✅ 0.21.0+cu124>=0.9.0',
  'tqdm': '✅ 4.67.1>=4.64.0',
  'psutil': '✅ 5.9.5',
  'py-cpuinfo': '✅ 9.0.0',
  'pandas': '✅ 2.2.2>=1.1.4',
  'seaborn': '✅ 0.13.2>=0.11.0',
  'ultralytics-thop': '✅ 2.0.14>=2.0.0'}}

### Minimal Reproducible Example

See StackOverflow link or the included code.

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @hack-r, thank you for bringing this to our attention 🚀! We appreciate your detailed report and effort in documenting this issue, including the StackOverflow reference.

To help us investigate further, could you please confirm the following details or provide additional clarification? 

1. **Minimum Reproducible Example (MRE)**: While you've shared a comprehensive code snippet, it would be helpful to isolate the issue further by providing a simplified version of the code that reproduces the problem. For guidance, refer to our [Minimum Reproducible Example](https://docs.ultralytics.com/help/minimum-reproducible-example/) documentation.
2. **Weight Path Validation**: Ensure that the specified weight file (e.g., `yolo12x.pt`) exists at the defined path and is not corrupted. If possible, re-download it to verify.
3. **Environment Validation**: Confirm that the issue persists after upgrading to the latest `ultralytics` version and dependencies. You can upgrade your environment with:
   ```bash
   pip install -U ultralytics
   ```

## Documentation and Resources

For additional guidance:
- Explore the [Train Settings](https://docs.ultralytics.com/modes/train/#train-settings) for clarification on parameter usage.
- Review our [Ultralytics HUB](https://hub.ultralytics.com/) to manage and monitor your experiments effortlessly.

## Verified Environments

To ensure compatibility, you can also verify your issue in one of our validated environments:
- **Notebooks**: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Cloud Solutions**: [Google Cloud](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/), [AWS](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/), and [Docker](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>.

## Community Support

Engage with the Ultralytics community for additional help:
- [Discord](https://discord.com/invite/ultralytics) 🎧 for real-time discussions.
- [Discourse](https://community.ultralytics.com/) for in-depth Q&A.
- [Reddit](https://reddit.com/r/Ultralytics) for community-driven insights.

Rest assured, this is an automated response 🤖, and an Ultralytics engineer will review and assist with your issue shortly. Thank you for your patience and understanding! 😊
